### PR TITLE
Streamline Cloud login with API tokens

### PR DESCRIPTION
**Pull request by jlowin**
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [x] adds new tests (if appropriate)
- [x] updates `CHANGELOG.md` (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?
This PR streamlines the process of logging in to Cloud and storing API tokens. Previously, `Client.login()` was a deprecated call that used old email/password auth. Now, `Client.login()` accepts an API token as its sole argument, and writes it to a known location. If the token is found in that location, it will be used automatically. 

This PR also removes token refresh behavior, as API tokens don't need refreshing.

### CODE REVIEW

File: src/prefect/client/client.py [hunk 1]
```diff
@@ -64,37 +64,31 @@ class Client:
...
+                    self.token_is_local = True
 
         self.token = token
 
+    @property
+    def local_token_path(self) -> str:
+        """
+        Returns the local token path corresponding to the provided graphql_server
+        """
+        graphql_server = (self.graphql_server or "").replace("/", "_")
```
Comment (cicdw) [inline src/prefect/client/client.py]: oh this is beautiful, thank you for this

File: src/prefect/client/client.py [hunk 2]
```diff
@@ -202,114 +196,66 @@ def _request(
...
-            raise AuthorizationError("Could not log in.")
-        self.token = response.json().get("token")
-        if self.token:
-            creds_path = os.path.expanduser("~/.prefect/.credentials")
-            if not os.path.exists(creds_path):
-                os.makedirs(creds_path)
-            with open(os.path.join(creds_path, "auth_token"), "w+") as f:
-                f.write(self.token)
+        if not os.path.exists(self.local_token_path):
+            os.makedirs(self.local_token_path)
```
Comment (cicdw) [inline src/prefect/client/client.py]: So the `local_token_path` appears to include the filename as well -> have you verified that this `makedirs` call doesn't accidentally make a _directory_ with the name of the intended token file?

Comment (jlowin) [reply src/prefect/client/client.py]: Excellent call, no

Comment (jlowin) [reply src/prefect/client/client.py]: Fixed and added test!

Comment (cicdw) [reply src/prefect/client/client.py]: Thanks!

**Pull request merged**